src/tests/Microsoft.PowerFx.Core.Tests/DisplayNameTests.cs

```diff
@@ -18,6 +20,80 @@ namespace Microsoft.PowerFx.Interpreter.Tests
 {
     public class DisplayNameTests : PowerFxTest
     {
+        public class LazyRecordType : RecordType
+        {
+            public override IEnumerable<string> FieldNames { get; }
+
+            public override bool TryGetFieldType(string name, out FormulaType type)
+            {
+                type = name switch
+                {
+                    "Num" => FormulaType.Number,
+                    "B" => FormulaType.Boolean,
+                    "Nested" => TableType.Empty(),
```

Comment:
This still isn't quite right. I suppose it's working because 
Sum([], number) is still valid. 
Could you add this test case? It would show the difference between the two, I think.

        [InlineData("Sum(NestedDisplay, ThisRecord.InnerDisplay)", "Sum(Nested, ThisRecord.Inner)", false)]